bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java

```diff
@@ -338,23 +394,80 @@ private VenstarAwayMode getAwayMode() {
         return infoData.getAway();
     }
 
-    private void updateSettings(VenstarAwayMode away) {
+    private VenstarFanMode getFanMode() {
+        return infoData.getFanMode();
+    }
+
+    private VenstarFanState getFanState() {
+        return infoData.getFanState();
+    }
+
+    private VenstarScheduleMode getScheduleMode() {
+        return infoData.getScheduleMode();
+    }
+
+    private VenstarSchedulePart getSchedulePart() {
+        return infoData.getSchedulePart();
+    }
+
+    private ZonedDateTime getTimestampRuntime(VenstarRuntime runtime) {
+        ZonedDateTime z = ZonedDateTime.ofInstant(Instant.ofEpochSecond(runtime.getTimeStamp()),
+                ZoneId.systemDefault());
```

Comment:
Actually it looks like this is reported in the user's timezone (or at least the time zone the device is set in) , so no need to convert to the system default time zone (or use GMT if you have to pass a timezone) .  But this is actually more tricky as the runtime schedules are more complicated, i'll post that in another comment.

Reply:
I was struggling with this due to lack of familiarity with timestamps, epoch, etc (I'm really an amateur....). What should I put as the function argument to Instant.ofEpochSecond() instead of "ZoneId.systemDefault()"? It's very confusing. I also noticed the time displayed in the channel was 2 hours ahead of my home time zone.

Reply:
So this still needs to be resolved , or the time is always going to be off.

So this is kinda tricky, but i think you need to offset this back to UTC and then create a ZoneDate time.  Dealing with time zones is always painful.   

So maybe something like....
```java
ZoneId  zoneId = ZoneId.systemDefault();
ZonedDateTime now = LocalDateTime.now().atZone(zoneId);
int diff = now.getOffset().getTotalSeconds();
ZonedDateTime z = ZonedDateTime.ofInstant(Instant.ofEpochSecond(runtime.getTimeStamp() -diff), zoneId);
```

```